samples/hybrid-connections/java/simple-websocket-demo/src/samples/WebsocketListener.java

```diff
@@ -0,0 +1,48 @@
+package samples;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import com.microsoft.azure.relay.ClientWebSocket;
+import com.microsoft.azure.relay.HybridConnectionListener;
+import com.microsoft.azure.relay.StringUtil;
+import com.microsoft.azure.relay.TokenProvider;
+
+public class WebsocketListener {
+	static boolean quit = false;
+	
+	static final String CONNECTION_STRING_ENV_VARIABLE_NAME = "RELAY_CONNECTION_STRING_ENVIRONMENT_VARIABLE";
+	static final Map<String, String> connectionParams = StringUtil.parseConnectionString(System.getenv(CONNECTION_STRING_ENV_VARIABLE_NAME));
+	static final String RELAY_NAME_SPACE = connectionParams.get("Endpoint");
+	static final String CONNECTION_STRING = connectionParams.get("EntityPath");
+	static final String KEY_NAME = connectionParams.get("SharedAccessKeyName");
+	static final String KEY = connectionParams.get("SharedAccessKey");
+	
+	public static void main(String[] args) throws URISyntaxException {
+		TokenProvider tokenProvider = TokenProvider.createSharedAccessSignatureTokenProvider(KEY_NAME, KEY);
+		HybridConnectionListener listener = new HybridConnectionListener(new URI(RELAY_NAME_SPACE + CONNECTION_STRING), tokenProvider);
+        
+        listener.openAsync().join();
+        while (!quit) {
+        	ClientWebSocket websocket = listener.acceptConnectionAsync().join();
+			System.out.println("New session connected.");
+			while (websocket.isOpen()) {
+				System.out.println("socket is open");
+				ByteBuffer bytesReceived = websocket.receiveMessageAsync().join();
+				System.out.println("msg received");
+				String msg = new String(bytesReceived.array());
+				if (msg.equals("quit")) {
+					System.out.println("quit!");
+					quit = true;
+					break;
+				}
+				System.out.println("Received: " + msg);
+				websocket.sendAsync("Echo: " + msg);
+			}
```

Comment:
Can we make it so that each client is processed at least by another thread so the listener can accept more than 1 client at a time?

We definitely don't want people to grab our sample and end up with a "highly scalable" listener which can only handle 1 concurrent connection.

Reply:
Listeners may have hundreds or thousands of senders.

---
In reply to: [242661300](https://github.com/Azure/azure-relay/pull/35#discussion_r242661300) [](ancestors = 242661300)